cat/ptx-proxy.cat

```diff
@@ -0,0 +1,81 @@
+PTX
+(* NVIDIA. Parallel Thread Execution ISA Version 7.5 *)
+
+(*******************)
+(*   Auxiliaries   *)
+(*******************)
+(* Events *)
+let strong-write = W & (RLX | REL)
+let strong-read = R & (RLX | ACQ)
+let strong-m = strong-write | strong-read
+let strong-operation = strong-write | strong-read | F
+
+(* Proxy *)
+let generic-proxy = GEN * GEN
+let surface-proxy = SUR * SUR
+let texture-proxy = TEX * TEX
+let constant-proxy = CON * CON
+let same-proxy = generic-proxy | surface-proxy | texture-proxy | constant-proxy
+
+(* Relations *)
+let sr-fence = ([strong-operation]; (sr & same-proxy); [F]) | ([F]; (sr & same-proxy); [strong-operation])
+let po-fence = ([strong-operation]; (po & same-proxy); [F]) | ([F]; (po & same-proxy); [strong-operation])
+let sr-m = ([strong-m]; (sr & loc & same-proxy); [strong-m])
+let po-m = ([M]; (po & loc & same-proxy); [M])
+let morally-strong = sr-fence | po-fence | sr-m | po-m
+(*******************)
+let fence-sc = [F & SC] ; sr; [F & SC]
+let rec observation = (morally-strong & rf) | (observation; rmw; observation)
```

Comment:
Why is it not `observation = (morally-strong & rf) | rmw`? 
I assume it is related to this comment, but I don't get it
```
/* There are two definitions: the original version was the raw definition
 * listed first, but PTX uses a free relation defined recursively, as shown
 * second.  The two should be equivalent. */
fun observation : Op->Op {
  strong[rf] + rmw
}
// fact { observation in ^(strong[rf] + rmw) }
// fact { strong[rf] in observation }
// fact { observation.rmw.observation in observation }
```

Reply:
Recursion is expensive for us, thus if both definition are equivalent, use the one with no recursion

Reply:
Accepted

Reply:
Now that I think again, we should use the non-recursive one but add a comment saying "this is equivalent to let rec observation = (morally-strong & rf) | (observation; rmw; observation)". This was every time we add new test cases we can actually check if we really get the same results no matter with of the two definitions we use

Reply:
Comment added to both `ptx.cat` and `ptx-proxy.cat`.